### PR TITLE
feat: add RailException support and improve error handling

### PR DESCRIPTION
**Pull request by Pouyanpi**
- implement RailException for injection detection (a must have for checks)
- Add TypedDict for structured return values
- improve error handling for malformed YARA rules

## Summary by GH Copilot

This pull request enhances the injection detection system by introducing a structured return type, improving error handling, and adding new tests for edge cases. The changes aim to make the system more robust and easier to integrate into workflows.

### Code Enhancements

* Introduced a `TypedDict` named `InjectionDetectionResult` to standardize the return type for injection detection functions, containing `is_injection`, `text`, and `detections` fields. (`nemoguardrails/library/injection_detection/actions.py`, [nemoguardrails/library/injection_detection/actions.pyR52-R57](diffhunk://#diff-2e8e7fb15761a3d7ce33b9010d2c4e1c7ae19dc092a5747dfeb1c70653695219R52-R57))
* Updated `_omit_injection`, `_sanitize_injection`, and `_reject_injection` functions to return structured tuples or lists for better clarity and integration. (`nemoguardrails/library/injection_detection/actions.py`, [[1]](diffhunk://#diff-2e8e7fb15761a3d7ce33b9010d2c4e1c7ae19dc092a5747dfeb1c70653695219L219-R236) [[2]](diffhunk://#diff-2e8e7fb15761a3d7ce33b9010d2c4e1c7ae19dc092a5747dfeb1c70653695219L256-R292) [[3]](diffhunk://#diff-2e8e7fb15761a3d7ce33b9010d2c4e1c7ae19dc092a5747dfeb1c70653695219L280-R306)

### Improved Error Handling

* Enhanced error messages for YARA rule compilation failures to provide more descriptive feedback. (`nemoguardrails/library/injection_detection/actions.py`, [nemoguardrails/library/injection_detection/actions.pyL200-R212](diffhunk://#diff-2e8e7fb15761a3d7ce33b9010d2c4e1c7ae19dc092a5747dfeb1c70653695219L200-R212))
* Ensured graceful handling of malformed YARA rules, logging errors without causing application crashes. (`tests/test_injection_detection.py`, [tests/test_injection_detection.pyR623-R768](diffhunk://#diff-e3377e30bb9064a790bca4ea01e34ba87d42ae5af0f8177d547fd17d4d4460c5R623-R768))

### Flow Updates

* Updated the injection detection flow logic in `.co` files to use the new `InjectionDetectionResult` format, improving clarity and handling of different actions (`reject`, `omit`, `sanitize`). (`nemoguardrails/library/injection_detection/flows.co`, [[1]](diffhunk://#diff-cfeacfe5cf452e3d537bbb9a0b3da63104b45bd8b6a100fc7ea681ffd1e2e5feL1-R19); `nemoguardrails/library/injection_detection/flows.v1.co`, [[2]](diffhunk://#diff-5222021e54fc2f84694ba1a94eb487d0d1918049262f85c9298a7443c3f9fff5L1-R19)

### Testing Improvements

* Added new test cases to validate edge scenarios, such as exceptions for specific actions, malformed YARA rules, and behavior when exceptions are enabled. (`tests/test_injection_detection.py`, [tests/test_injection_detection.pyR623-R768](diffhunk://#diff-e3377e30bb9064a790bca4ea01e34ba87d42ae5af0f8177d547fd17d4d4460c5R623-R768))
* Refactored existing tests to align with the updated function signatures and return types. (`tests/test_injection_detection.py`, [[1]](diffhunk://#diff-e3377e30bb9064a790bca4ea01e34ba87d42ae5af0f8177d547fd17d4d4460c5L268-R269) [[2]](diffhunk://#diff-e3377e30bb9064a790bca4ea01e34ba87d42ae5af0f8177d547fd17d4d4460c5L410-R416)

### CODE REVIEW

File: nemoguardrails/library/injection_detection/actions.py [hunk 1]
```diff
@@ -289,18 +314,20 @@ def _reject_injection(text: str, rules: "yara.Rules") -> Tuple[bool, str]:
...
-        return False, ""
+        return False, []
     matches = rules.match(data=text)
     if matches:
-        matches_string = ", ".join([match_name.rule for match_name in matches])
-        log.info(f"Input matched on rule {matches_string}.")
-        return True, matches_string
+        matched_rules = [match_name.rule for match_name in matches]
+        log.info(f"Input matched on rule {', '.join(matched_rules)}.")
+        return True, matched_rules
```
Comment (erickgalinkin) [inline nemoguardrails/library/injection_detection/actions.py:322]: This is probably much cleaner logic. Solid.

File: nemoguardrails/library/injection_detection/actions.py [hunk 2]
```diff
@@ -310,45 +337,68 @@ async def injection_detection(text: str, config: RailsConfig) -> str:
...
+            else:
+                raise NotImplementedError(
+                    f"Expected `action` parameter to be 'reject', 'omit', or 'sanitize' but got {action_option} instead."
+                )
+        # no matches found
         else:
-            # We should never ever hit this since we inspect the action option above, but putting an error here anyway.
-            raise NotImplementedError(
-                f"Expected `action` parameter to be 'omit' or 'sanitize' but got {action_option} instead."
+            return InjectionDetectionResult(
```
Comment (erickgalinkin) [inline nemoguardrails/library/injection_detection/actions.py:402]: I'm into this.

**Review by erickgalinkin — APPROVED** [review]
This looks good to me, thanks!

**Review by cparisien — APPROVED** [review]
